<a href="https://colab.research.google.com/github/baldpanda/mnist/blob/master/mnist.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import tensorflow as tf
print(tf.VERSION)

1.13.1


Keras comes with a number of inbuilt datasets. These include:

*   CIFAR10 AND CIFAR100 image classification (50,000 images with 10 and 100 different classes)
*   MNIST
*   Fashion MNIST
*   Boston Housing Regression Dataset




In [0]:
mnist = tf.keras.datasets.mnist

In [0]:
(X_train, y_train), (X_test, y_test) = mnist.load_data()

In [17]:
print('The training data has shape:{}.'.format(X_train.shape))

The training data has shape:(60000, 28, 28).


In [18]:
print('The testing data has size: {}.'.format(X_test.shape[0]))

The testing data has size: 10000.


Initially building a simple fully connected neural network to train on the MNIST dataset.

In [0]:
X_train = X_train / 255.0
X_test = X_test / 255.0

In [0]:
model = tf.keras.Sequential([
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(128, activation = tf.nn.relu),
    tf.keras.layers.Dense(10, activation = tf.nn.softmax)
])

In [21]:
model.compile(optimizer=tf.train.AdamOptimizer(),
             loss='sparse_categorical_crossentropy',
             metrics=['accuracy'])

Instructions for updating:
Colocations handled automatically by placer.


In [22]:
model.fit(X_train, y_train, epochs=5)

Epoch 1/5
60000/60000 [==============================] - 6s 97us/sample - loss: 0.2542 - acc: 0.9286
Epoch 2/5
60000/60000 [==============================] - 5s 85us/sample - loss: 0.1115 - acc: 0.9664
Epoch 3/5
60000/60000 [==============================] - 5s 83us/sample - loss: 0.0761 - acc: 0.9769
Epoch 4/5
60000/60000 [==============================] - 5s 79us/sample - loss: 0.0570 - acc: 0.9827
Epoch 5/5
60000/60000 [==============================] - 5s 81us/sample - loss: 0.0444 - acc: 0.9864


In [23]:
model.evaluate(X_test, y_test)

10000/10000 [==============================] - 1s 51us/sample - loss: 0.0751 - acc: 0.9775


[0.0750749997597537, 0.9775]

Building a basic 2 layer fully connected NN that achieves 97.8% accuracy on the test data. 